### ISSUE TITLE
char *api_get_nick(void)

### ISSUE THREAD

**Issue by lsatenstein** [JFreegman/toxic#523]
The code should be checking for len variable greater than zero.

when len == 0 you malloc 1 byte, 
name will not be null, but will likely be corrupted.

**Comment by JFreegman** [comment]
If `len` is 0 then `tox_self_get_name` has no effect per the API documentation. As a result, the `name` buffer will be a 1 byte buffer containing a null byte.

**Comment by lsatenstein** [comment]
Hi John
Thanks for your reply. It's a nice offering. 
In that source, should you be testing the Len var and return NULL  if it is zero?  
I don't know your API so do excuse my comments.  
Your makefile does not have all $variables defined. I like clean compiles and mistakenly thought you would want a third eye review of the code.That's what FOSS is about, right?


Sent from Yahoo Mail on Android 
 
  On Tue., 22 Jan. 2019 at 2:32 p.m., JFreegman<notifications@github.com> wrote:   
If len is 0 then tox_self_get_name has no effect per the API documentation. As a result, the name buffer will be a 1 byte buffer containing a null byte.

**Comment by JFreegman** [comment]
>In that source, should you be testing the Len var and return NULL  if it is zero?  

It's not necessary as I explained.

>Your makefile does not have all $variables defined. I like clean compiles and mistakenly thought you would want a third eye review of the code.That's what FOSS is about, right?

We can never have enough pairs of eyes looking over the code. You might want to familiarize yourself with the way the [Tox API ](https://github.com/TokTok/c-toxcore/blob/master/toxcore/tox.h#L1160)works though, because a lot of the code won't make much sense otherwise.

**Comment by lsatenstein** [comment]
Hi JFI am marking up a git branch of  one source. Just as some people like crossword puzzles, I like to review source code. 
About 10 years ago before I retired we wrote some secure file transfer. I retired and abondoned the marketing and closed the company. . At the time we used. A USB based TPM for sender/recipient. The software may still be used by some financial institutions. 
Anyway, I am not a competitor, just a keen applied encryption enthusiast. 
 Are you a he or a she? J can be John, Jed, Julia, Josh Jennifer or .....
Sent from Yahoo Mail on Android 
 
  On Tue., 22 Jan. 2019 at 6:08 p.m., JFreegman<notifications@github.com> wrote:   

In that source, should you be testing the Len var and return NULL if it is zero?


It's not necessary as I explained.


Your makefile does not have all $variables defined. I like clean compiles and mistakenly thought you would want a third eye review of the code.That's what FOSS is about, right?


We can never have enough pairs of eyes looking over the code. You might want to familiarize yourself with the way the Tox API works though, because a lot of the code won't make much sense otherwise.
